### ISSUE TITLE
CommandMap is Broken on Paper 1.15.2

### ISSUE THREAD

**Issue by ghost** [games647/CommandForward#3]
Running `/forward Alexis_Evelyn bungee` from chat causes the message `Error occurred executing command commandMap`. No errors show up in console when running this command.

Running `/forward bridgePlayer ping` from chat causes the generic error message `An internal error occurred while attempting to perform this command`. The following error shows up in console when running this command.

```
[16:56:15 ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing command 'forward' in plugin CommandForward v0.3.1
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[patched_1.15.2.jar:git-Paper-77]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[patched_1.15.2.jar:git-Paper-77]
        at org.bukkit.craftbukkit.v1_15_R1.CraftServer.dispatchCommand(CraftServer.java:740) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.PlayerConnection.handleCommand(PlayerConnection.java:1820) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:1628) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.PacketPlayInChat.a(PacketPlayInChat.java:47) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:23) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:1038) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:1031) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:1015) ~[patched_1.15.2.jar:git-Paper-77]
        at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:938) ~[patched_1.15.2.jar:git-Paper-77]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_231]
Caused by: java.lang.NullPointerException
        at com.github.games647.commandforward.bukkit.CommandForwardBukkit.onCommand(CommandForwardBukkit.java:52) ~[?:?]
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[patched_1.15.2.jar:git-Paper-77]
        ... 17 more
```

Edit: I am using [release 0.3.1](https://github.com/games647/CommandForward/releases/tag/0.3.1) on both bungee and paper.

Edit 2: **The plugin works only when the player is not op. Also, it's only commands like `/forward Alexis_Evelyn ban Alexis_Evelyn` that work (after giving the required perm first). `/forward playerBridge bungee` is still broken with the NullPointerException.**

Edit 3: Tested Commands:
* /forward non-existent-player ban Alexis_Evelyn - Does Not Work No Matter What (NullPointerException on bukkit side)
* /forward playerName bungee - Only Works if Player is Not OP (Command Blocks Also Fail For Same Reason)
* /forward console bungee - Only Works if Player is Not OP (Command Blocks Also Fail For Same Reason)

**Comment by ghost** [comment]
~~I believe the null pointer exception is coming from this line of code:~~ ~~https://github.com/games647/CommandForward/blob/6ecc823de0272bfa5e67d24d4d63225eba41548b/bukkit/src/main/java/com/github/games647/commandforward/bukkit/CommandForwardBukkit.java#L33~~

~~Edit: I never get the message "Player not online for forwarding this command" no matter what player I choose, just the generic error message.~~

Nvm, I just realized that this finds a random player to pick. This part works fine. I tested from console.

**Comment by ghost** [comment]
The command map issue occurs on the bungee side of the plugin.

```
[17:44:58 WARN] [CommandForward]: Cannot access command map for executing command
java.lang.NoSuchFieldException: commandMap
        at java.lang.Class.getField(Class.java:1703) ~[?:1.8.0_231]
        at com.github.games647.commandforward.bungee.CommandForwardBungee.invokeCommand(CommandForwardBungee.java:66) ~[?:?]
        at com.github.games647.commandforward.bungee.CommandForwardBungee.parseMessage(CommandForwardBungee.java:58) ~[?:?]
        at com.github.games647.commandforward.bungee.CommandForwardBungee.onServerConnected(CommandForwardBungee.java:44) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor10.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_231]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_231]
        at net.md_5.bungee.event.EventHandlerMethod.invoke(EventHandlerMethod.java:19) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at net.md_5.bungee.event.EventBus.post(EventBus.java:46) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at net.md_5.bungee.api.plugin.PluginManager.callEvent(PluginManager.java:435) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at net.md_5.bungee.connection.DownstreamBridge.handle(DownstreamBridge.java:269) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at net.md_5.bungee.protocol.packet.PluginMessage.handle(PluginMessage.java:102) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at net.md_5.bungee.netty.HandlerBoss.channelRead(HandlerBoss.java:104) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:321) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:295) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:792) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:475) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[Waterfall.jar:git:Waterfall-Bootstrap:1.15-SNAPSHOT:4339bb7:317]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_231]
```

The null pointer reference appears to execute on the bukkit side of the plugin.

**Comment by ghost** [comment]
The only thing I didn't fix is the op permission bypass. I just disabled OP bypass completely. I'm currently working on adding a permission node to being able to run a command as console or another user. That way mods cannot just bypass the permission system when they ban someone using the ban gui. I'll send in a pull request after I'm completely finished.

**Comment by bloggy** [comment]
still not working on 1.15.2 for me

**Comment by TuxCoding** [comment]
Fixed by #9
